EIPS/eip-7892.md

```diff
@@ -59,37 +71,107 @@ To facilitate these changes on the execution layer, the `blobSchedule` object sp
     "max": 9,
     "baseFeeUpdateFraction": 5007716
   },
-  "1740693335": {
+  "12000000": {
     "target": 24,
     "max": 48,
     "baseFeeUpdateFraction": 5007716
   },
-  "1743285335": {
+  "12300000": {
     "target": 36,
     "max": 56,
     "baseFeeUpdateFraction": 5007716
   }
 }
 ```
 
-On the consensus layer, a new parameter is added to the configuration:
+### Consensus layer configuration
 
-```
+A new `BLOB_SCHEDULE` field is added to consensus layer configuration, containing a sequence of entries representing blob parameter changes **after** `ELECTRA_FORK_EPOCH`. There exists one entry per fork that changes blob parameters, whether it is a regular or a Blob-Parameter-Only fork.
+
+```yaml
 BLOB_SCHEDULE:
-  - EPOCH: 348618
+  - EPOCH: 400000     ## A future anonymous BPO fork
     MAX_BLOBS_PER_BLOCK: 24
-  - EPOCH: 355368
+  - EPOCH: 420000     ## A future anonymous BPO fork
     MAX_BLOBS_PER_BLOCK: 56
+  - EPOCH: 440000     ## GLOAS_FORK_EPOCH; a future named fork introducing blob parameter changes
+    MAX_BLOBS_PER_BLOCK: 72
 ```
 
 The parameters and schedules above are purely illustrative. Actual values and schedules are beyond the scope of this specification.
 
-### Requirements
+**Requirements:**
+
+- Execution and consensus clients **MUST** share consistent BPO fork schedules.
+- The slot number in the EL's `blobSchedule` **MUST** align with the start of the epoch specified in the consensus layer configuration.
+- The `max` field in the EL's `blobSchedule` **MUST** equal the `MAX_BLOBS_PER_BLOCK` value in the consensus layer configuration.
+
+### Modified `compute_fork_digest`
+
+The `compute_fork_digest` helper is updated to account for BPO forks:
+
+```python
+# These basic types already exist. Copied here for reference.
+Epoch = int
+Version = bytes
+Root = bytes
+ForkDigest = bytes
+
+class BlobScheduleEntry(NamedTuple):
+   epoch: Epoch
+   max_blobs_per_block: int       # uint32
```

Comment:
Currently, `max_blobs_per_block` is a uint64 [in the specs](https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/beacon-chain.md#execution-1) though it could be a uint16 if we wanted, since [`MAX_BLOB_COMMITMENTS_PER_BLOCK`](https://jtraglia.github.io/eth-spec-viewer/#preset_vars-MAX_BLOB_COMMITMENTS_PER_BLOCK) is 4,096 and unlikely to ever be greater than 65,535.

Reply:
Alternatively we could hash the value and bitmask a slice of it, but IMO not worth it.

Reply:
Yeah this is tricky. I suppose an extra assert would be fine.